packages/amplify-schema-validator/src/validators/fields-passed-are-in-parent-model.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  DocumentNode,
+  Kind,
+  ListValueNode,
+  ObjectTypeDefinitionNode,
+  StringValueNode,
+} from 'graphql';
+import { InvalidDirectiveError } from '../exceptions/invalid-directive-error';
+import { getObjectWithName } from '../helpers/get-object-with-name';
+import { resolveFieldTypeName } from '../helpers/resolve-field-type-name';
+
+/**
+   * Validates that any of the fields passed in are in the parent model
+   *
+   * @param schema graphql schema
+   * @returns true
+   */
+export const validateFieldInParentModel = (schema: DocumentNode): Error[] => {
+  const errors: Error[] = [];
+  const objectTypeDefinitions = schema.definitions.filter(
+    (defintion) => defintion.kind === Kind.OBJECT_TYPE_DEFINITION,
+  ) as ObjectTypeDefinitionNode[];
+  objectTypeDefinitions.forEach((objectTypeDefinition) => {
+    const directiveFields = objectTypeDefinition.fields?.filter((objectField) => objectField.directives?.find(
+      (directive) => directive.name.value,
+    ));
+
+    directiveFields?.forEach((directiveField) => {
+      const typeName = resolveFieldTypeName(directiveField.type);
+      if (!typeName) {
+        /* istanbul ignore next */
+        return;
+      }
+      const objectOfType = getObjectWithName(schema, typeName);
+      const fields = objectOfType?.fields;
+      if (!fields) {
+        /* istanbul ignore next */
+        return;
+      }
+      const fieldVals: string[] = [];
+      for (let i = 0; i < fields?.length; i++) {
+        const val = fields[i].name.value;
+        fieldVals.push(val);
+      }
```

Comment:
This can be simplified with array.map function.

Reply:
Fixed